packages/commonjs/src/resolve-require-sources.js

```diff
@@ -196,16 +196,21 @@ export function getRequireResolver(extensions, detectCyclesAndConditional, curre
           // Special-case external Node built-ins to be handled via a lazy __require
           // helper instead of hoisted ESM imports when strict wrapping is used.
           const isExternalWrapped = isWrappedId(dependencyId, EXTERNAL_SUFFIX);
-          if (
-            parentMeta.initialCommonJSType === IS_WRAPPED_COMMONJS &&
-            !allowProxy &&
-            isExternalWrapped
-          ) {
+          let resolvedDependencyId = dependencyId;
+          if (parentMeta.isCommonJS === IS_WRAPPED_COMMONJS && !allowProxy && isExternalWrapped) {
```

Comment:
1. Previously, we checked `parentMeta.initialCommonJSType === IS_WRAPPED_COMMONJS` instead of `parentMeta.isCommonJS === IS_WRAPPED_COMMONJS`. With strictRequires: 'auto', I believe the initial type (before cycle detection) can differ from the final type.